include/stdexec/execution.hpp

```diff
@@ -5411,11 +5432,19 @@ namespace stdexec {
             __ignore>,
           _Senders...>;
 
-      using __errors_variant = //
+      using __error_types = //
         __minvoke<
-          __mconcat<__q<__variant>>,
-          __types<std::exception_ptr>,
-          error_types_of_t< _Senders, __env_t<__id<_Env>>, __types>...>;
+          __mconcat<__transform<__q<decay_t>, __munique<>>>,
+          error_types_of_t<_Senders, __env_t<__id<_Env>>, __types>... >;
+
+      using __errors_variant = //
+        __if<
+          __all_value_and_error_args_nothrow_decay_copyable<_Env, __id<_Senders>...>,
+          __if<
+            std::is_same<__error_types, __types<>>,
+            std::variant<std::monostate>,
+            __mapply<__q<std::variant>, __error_types>>,
+          __minvoke<__push_back_unique<__q<std::variant>>, __error_types, std::exception_ptr>>;
```

Comment:
Probably best to just use `__nullable_variant` here and always have the `monostate` alternative first. Then you don't have to wrap the variant in an optional above.

Reply:
Ok I did as you asked, but that had its own problems. I wasnt able to detect whether `std::monostate` is a valid error type and I had to manually ignore it as an error type. Constraining on `__callable<_Tag, _Receiver&&, std::monostate>` didn't work as expected, probably due to a bug in constrains of `schedule_from`s receiver. I couldnt solve that one.

Reply:
I suppose you could avoid that problem by using an unspecified, non-public type instead of std::monostate.

Reply:
I think the cleanest solution is manual lifetime but I will go with the private monostate type until I sneaked in the former.